include/hipSYCL/sycl/libkernel/sscp/builtins.hpp

```diff
@@ -164,6 +164,17 @@ HIPSYCL_BUILTIN double __hipsycl_frexp(double x, IntT* ptr) {
 
 HIPSYCL_DEFINE_SSCP_GENFLOAT_MATH_BUILTIN2(hypot)
 HIPSYCL_DEFINE_SSCP_GENFLOAT_MATH_BUILTIN(ilogb)
+
+template<class IntType>
+HIPSYCL_BUILTIN float __hipsycl_ldexp(float x, IntType k) noexcept {
+  return __hipsycl_sscp_ldexp_f32(x, k);
```

Comment:
```suggestion
  return __hipsycl_sscp_ldexp_f32(x, static_cast<__hipsycl_int32>(k));
```
Do we need a cast here since IntType can be arbitrary, but the SSCP builtins only want one specific type of int?